Client/game_sa/CStreamingSA.cpp

```diff
@@ -402,7 +402,7 @@ unsigned char CStreamingSA::AddArchive(const char* szFilePath)
 
     // Create new stream handler
     const auto streamCreateFlags = *(DWORD*)0x8E3FE0;
-    HANDLE     hFile = CreateFileA(szFilePath, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING,
+    HANDLE     hFile = CreateFileW(szFilePath, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING,
```

Comment:
A `FromUTF8` would have been enough here.

Reply:
Doesn't work for me here:

```c++
unsigned char CStreamingSA::AddArchive(const char* szFilePath) {
//...
CreateFileW(FromUTF8(szFilePath), ...)
//...
}
```

Reply:
Maybe because FromUTF8 returns a SString and you need a pointer to wchar_t there? Try `*FromUTF8(...)`.

Reply:
FromUTF8 returns a WString

Reply:
Code compiles but FromUTF8 returns a string trimmed to first utf-8 character.

Reply:
btw, i didn't remove `std::wstring utf8Path = SharedUtil::MbUTF8ToUTF16(strFullPath);`
because it breaks std::filesystem api

Reply:
My point was, that you could probably reduce this pull request just to

![image](https://github.com/multitheftauto/mtasa-blue/assets/38703318/634314d3-f0a4-4409-9889-9da6cf43cc87)

Reply:
i thought so. But this doesn't work